### PR TITLE
Document `IN` operator

### PR DESCRIPTION
**Pull request by Ankan0011**
## Description
Add IN operator to the comparison section. This is a improvement for the documentation with examples.

## Additional context and related issues
Fixes https://github.com/trinodb/trino/issues/4791

## Release notes

(X) This is not user-visible or is docs only, and no release notes are required.
( ) Release notes are required. Please propose a release note for me.
( ) Release notes are required, with the following suggested text:

### CODE REVIEW

**Comment by ebyhr** [comment]
This is duplicate of #12810

File: docs/src/main/sphinx/functions/comparison.md [hunk 1]
```diff
@@ -256,3 +256,47 @@ you need to match the used escape character as well, you can escape it.
...
+... WHERE column [NOT] IN ('value1','value2');
+... WHERE column [NOT] IN ( subquery )
+```
+
+The `IN` operator can be use to fetch multiple records according to mutiple 
+values specified in WHERE clause. A sub-query or list of values must be specified 
+in the parenthesis.
+
+```
+SELECT * FROM region WHERE name IN ('America', 'Europe');
```
Comment (ebyhr) [inline docs/src/main/sphinx/functions/comparison.md:276]: The actual `name` value is uppercase. Same for others. 
```suggestion
SELECT * FROM region WHERE name IN ('AMERICA', 'EUROPE');
```

File: docs/src/main/sphinx/functions/comparison.md [hunk 2]
```diff
@@ -256,3 +256,47 @@ you need to match the used escape character as well, you can escape it.
...
+## Row comparison: IN
+
+The `IN` operator can be used to compare values with the following patterns:
+
+```
+... WHERE column [NOT] IN ('value1','value2');
+... WHERE column [NOT] IN ( subquery )
+```
+
+The `IN` operator can be use to fetch multiple records according to mutiple
```
Comment (mosabua) [inline docs/src/main/sphinx/functions/comparison.md:271]: Remove this sentence or move the content up

File: docs/src/main/sphinx/functions/comparison.md [hunk 3]
```diff
@@ -256,3 +256,47 @@ you need to match the used escape character as well, you can escape it.
...
+
+The `IN` operator can be use to fetch multiple records according to mutiple 
+values specified in WHERE clause. A sub-query or list of values must be specified 
+in the parenthesis.
+
+```
+SELECT * FROM region WHERE name IN ('AMERICA', 'EUROPE');
+```
+
+It is shorthand of mutiple `OR` operator. The same query can be modified to
```
Comment (mosabua) [inline docs/src/main/sphinx/functions/comparison.md:279]: Maybe better to say
```
The values in the clause are used for multiple comparisons that are combined as a logical or. The preceding query is equivalent to the following query:
```

File: docs/src/main/sphinx/functions/comparison.md [hunk 4]
```diff
@@ -256,3 +256,47 @@ you need to match the used escape character as well, you can escape it.
...
+```
+
+It is shorthand of mutiple `OR` operator. The same query can be modified to 
+include a list of values seperated with `OR` operator.
+
+```
+SELECT * FROM region WHERE name = 'AMERICA' OR name = 'EUROPE';
+```
+
+You can also negate the result by adding `NOT`, and get all other regions except the values in list:
```
Comment (mosabua) [inline docs/src/main/sphinx/functions/comparison.md:286]:
```suggestion
You can negate the comparisons by adding `NOT`, and get all other regions except the values in list:
```

File: docs/src/main/sphinx/functions/comparison.md [hunk 5]
```diff
@@ -256,3 +256,47 @@ you need to match the used escape character as well, you can escape it.
...
+
+```
+SELECT * FROM region WHERE name NOT IN ('AMERICA', 'EUROPE');
+```
+
+You can also use the subquery with the IN operator that returns records from the single column.
```
Comment (mosabua) [inline docs/src/main/sphinx/functions/comparison.md:292]: When using a subquery to determine the values to use in the comparison, the subquery must return a single column and one or more rows.

File: docs/src/main/sphinx/functions/comparison.md [hunk 6]
```diff
@@ -256,3 +256,47 @@ you need to match the used escape character as well, you can escape it.
...
+
+You can also use the subquery with the IN operator that returns records from the single column. 
+The subquery cannot include more than one column in the SELECT column list.
+
+```
+SELECT name FROM nation 
+WHERE regionkey IN ( 
+    SELECT regionkey 
+    FROM region 
+    WHERE name IN ('AMERICA', 'EUROPE')
```
Comment (mosabua) [inline docs/src/main/sphinx/functions/comparison.md:300]: modify to use name startswith A

**Review by mosabua — CHANGES_REQUESTED** [review]

**Comment by mosabua** [comment]
Please squash commits

**Comment by Ankan0011** [comment]
Messed up the PR. Creating a new PR for the same.

**Pull request closed without merging**